src/kernel/sporadic.c

```diff
@@ -167,7 +167,7 @@ void refill_new(sched_context_t *sc, word_t max_refills, ticks_t budget, ticks_t
     /* full budget available */
     refill_head(sc)->rAmount = budget;
     /* budget can be used from now */
-    refill_head(sc)->rTime = NODE_STATE_ON_CORE(ksCurTime, core);
+    refill_head(sc)->rTime = NODE_STATE(ksCurTime);
```

Comment:
Now that `core` is not used here any longer, It seems there is no reason to pass it to `refill_new()` at all? Removing this could happen in an additional commit, will also remove some of SMP-conditional compilation from the code

Reply:
Well spotted. I also removed the REFILL_NEW macro.